### PR TITLE
Expanded /years endpoint

### PR DESCRIPTION
**Pull request by jrm86**
Returns years as student if user is a student.  Returns years as instructor if user is an instructor.  Returns all years if else.

Also removed 'gradebook.' schema qualifier from other queries.

This change makes an assumption that the value of the userRole variable is provided by the UI.

### CODE REVIEW

File: src/webapp/gradebookServer.js [hunk 1]
```diff
@@ -160,11 +160,23 @@ app.get('/years', function(request, response) {
...
-   var queryParams = [instructorID];
+   if (userRole == 'alpha_GB_Instructor')
+   {
+      queryText = 'SELECT Year FROM getInstructorYears($1);';
+   } 
+   else if (userRole == 'alpha_GB_Student')
+   {
+      queryText = 'SELECT Year FROM getStudentYears($1);';
+   }
+   else { queryText = 'SELECT DISTINCT year FROM term;';
```
Comment (afig) [inline src/webapp/gradebookServer.js]:
```suggestion
   else
   {
      queryText = 'SELECT DISTINCT year FROM term;';
```
Formatting inconsistency

File: src/webapp/gradebookServer.js [hunk 2]
```diff
@@ -160,11 +160,23 @@ app.get('/years', function(request, response) {
...
    //Set the query text
-   var queryText = 'SELECT Year FROM gradebook.getInstructorYears($1);';
-   var queryParams = [instructorID];
+   if (userRole == 'alpha_GB_Instructor')
+   {
+      queryText = 'SELECT Year FROM getInstructorYears($1);';
+   } 
+   else if (userRole == 'alpha_GB_Student')
+   {
+      queryText = 'SELECT Year FROM getStudentYears($1);';
```
Comment (afig) [inline src/webapp/gradebookServer.js]:
```suggestion
      queryText = 'SELECT Year FROM getYearsAsStudent();';
```

Students do not have access to `getStudentYears();` and so they will not be able to use it. Instead, `getYearsAsStudent();` should be used.

File: src/webapp/gradebookServer.js [hunk 3]
```diff
@@ -160,11 +160,23 @@ app.get('/years', function(request, response) {
...
+   var userID = request.query.userID;
+   var userRole = request.query.userRole; //assuming this is provided at the UI level
+   var queryText;
 
    //Set the query text
-   var queryText = 'SELECT Year FROM gradebook.getInstructorYears($1);';
-   var queryParams = [instructorID];
+   if (userRole == 'alpha_GB_Instructor')
+   {
+      queryText = 'SELECT Year FROM getInstructorYears($1);';
```
Comment (afig) [inline src/webapp/gradebookServer.js:171]:
```suggestion
      queryText = 'SELECT Year FROM getInstructorYears($1);';
      queryParams = [userID];
```

Sets parameters to `queryParams` for the one query that needs it.

File: src/webapp/gradebookServer.js [hunk 4]
```diff
@@ -160,11 +160,23 @@ app.get('/years', function(request, response) {
...
    //Get the params from the url
-   var instructorID = request.query.instructorid;
+   var userID = request.query.userID;
+   var userRole = request.query.userRole; //assuming this is provided at the UI level
+   var queryText;
 
    //Set the query text
-   var queryText = 'SELECT Year FROM gradebook.getInstructorYears($1);';
-   var queryParams = [instructorID];
+   if (userRole == 'alpha_GB_Instructor')
```
Comment (afig) [inline src/webapp/gradebookServer.js:169]:
```suggestion
   var queryParams;
   if (userRole == 'alpha_GB_Instructor')
```

The `pg` module throws an error if too many parameters are supplied, which means that we will need to conditionally set parameters. To do so, `queryParams` should be declared before the conditional query selection.

**Review by afig — COMMENTED** [review]
Thanks for adding this in a timely manner @jrm86. Sorry I couldn't review this earlier. Overall, the changes look good. I made a few suggestions that are accompanied with some explanations.

Edit: Also, not noted below: I recommend updating the header comment of the block that implements the `/years` end point, since not all users instruct sections (e.g. students are enrolled in sections)

File: src/webapp/gradebookServer.js [hunk 5]
```diff
@@ -160,11 +160,27 @@ app.get('/years', function(request, response) {
...
+   else if (userRole == 'alpha_GB_Student')
+   {
+      queryText = 'SELECT Year FROM getYearsAsStudent();';
+   }
+   else
+   {
+      queryText = 'SELECT DISTINCT year FROM term;';
+   }
+
+   var queryParams = [userID];
```
Comment (afig) [inline src/webapp/gradebookServer.js]:
```suggestion
```
This is set in the conditional that comes before it, so no need to re-declare it.

**Review by afig — COMMENTED** [review]
Other than the comment just added, this looks good to merge.

**Review by DasilvaB — APPROVED** [review]
Everything looks good.

File: src/webapp/gradebookServer.js (hunk 5 above)
Comment (jrm86) [reply src/webapp/gradebookServer.js]: Good catch, thank you.

**Review by afig — APPROVED** [review]

**Pull request merged**